extension/injections/extensible_injector.cpp

```diff
@@ -0,0 +1,131 @@
+//
+// Copyright (c) 2012-2017 Kanstantsin Chernik
+//
+// Distributed under the Boost Software License, Version 1.0.
+// (See accompanying file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+//<-
+#include <cassert>
+#include <memory>
+//->
+#include <boost/di.hpp>
+
+namespace di = boost::di;
+
+template <class TDep>
+class dependency_proxy
+    : di::core::dependency_base,
+      public di::core::dependency_impl<di::core::dependency_concept<typename TDep::expected, typename TDep::name>,
+                                       dependency_proxy<TDep>> {
+ public:
+  using scope = typename TDep::scope;
+  using expected = typename TDep::expected;
+  using given = typename TDep::given;
+  using name = typename TDep::name;
+  using priority = typename TDep::priority;
+  using scope_t = typename scope::template scope<expected, given>;
+
+  dependency_proxy(TDep& orig_dep) noexcept : ref_dep_(orig_dep) {}
+
+  dependency_proxy(dependency_proxy& other) noexcept : ref_dep_(other.ref_dep_) {}
+  dependency_proxy(dependency_proxy&& other) noexcept : ref_dep_(other.ref_dep_) {}
+
+  template <class T>
+  using is_referable = typename scope_t::template is_referable<T>;
+
+  template <class T, class TName, class TProvider>
+  static decltype(scope_t::template try_create<T, TName>(di::aux::declval<TProvider>())) try_create(const TProvider&);
+
+  template <class T, class TName, class TProvider>
+  auto create(const TProvider& provider) {
+    return ref_dep_.template create<T, TName>(provider);
+  }
+
+ private:
+  TDep& ref_dep_;
+};
+
+template <class TDependency, class TInjector>
+auto make_extensible_impl(const di::aux::type<TDependency>&, TInjector* injector) {
```

Comment:
Did you consider not to changing `TInjector` to a pointer to be coherent?

Reply:
I just copy/pasted this code from core :) Will try

Reply:
Yes, Will push this improvement